src/crypto/byron/ledgerUtils.js

```diff
@@ -0,0 +1,351 @@
+// @flow
+
+import AppAda from '@cardano-foundation/ledgerjs-hw-app-cardano'
+import {BigNumber} from 'bignumber.js'
+import TransportBLE from '@ledgerhq/react-native-hw-transport-ble'
+
+import {Logger} from '../../utils/logging'
+import {CONFIG} from '../../config'
+import {
+  generateFakeWallet,
+  signTransaction,
+  encodeTxAsRust,
+  decodeRustTx,
+} from './util'
+import {Bip32PublicKey} from 'react-native-chain-libs'
+
+import type {
+  Addressing,
+  PreparedTransactionData,
+  TransactionInput,
+  TransactionOutput,
+  TxBodiesResponse,
+  V1SignedTx,
+} from '../../types/HistoryTransaction'
+import type {
+  GetVersionResponse,
+  GetExtendedPublicKeyResponse,
+  InputTypeUTxO,
+  OutputTypeAddress,
+  OutputTypeChange,
+  SignTransactionResponse,
+  Witness,
+} from '@cardano-foundation/ledgerjs-hw-app-cardano'
+import type {TxWitness} from './util'
+import type BluetoothTransport from '@ledgerhq/react-native-hw-transport-ble'
+
+// these are defined in LedgerConnectStore.js in yoroi-frontend
+type LedgerConnectionResponse = {
+  versionResp: GetVersionResponse,
+  extendedPublicKeyResp: GetExtendedPublicKeyResponse,
+  deviceId: string,
+}
+
+type LedgerSignTxPayload = {
+  inputs: Array<InputTypeUTxO>,
+  outputs: Array<OutputTypeAddress | OutputTypeChange>,
+}
+
+// Hardware wallet device Features object
+// borrowed from HWConnectStoreTypes.js in yoroi-frontend
+export type HWFeatures = {
```

Comment:
I removed a lot of these from the Yoroi Extension. Notably, I moved it down to 

```
export type HWFeatures = {|
  Vendor: string,
  Model: string,
  DeviceId: string,
|};
```
but I also use Label for the default name.
However, not that almost all of these are Trezor only (I think `vendor` is the only one provided by Ledger)

Reply:
Agree. I'll simplify. But why removing the app version? I think it's important if we need to debug any issue eventually (even though there should be very few version upgrades, I guess)

Reply:
The app version doesn't really make sense to store because it can change at any time. Somebody can start Yoroi, then while the app is still running, do an update of their Ledger device. If you ever need to do something based off the version, you would need to query it at the time